analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -233,14 +242,35 @@ class Cargo(
             { extractPackage(it, hashes) }
         )
 
-        val dependencies = resolveDependencyTree(projectName, projectVersion, metadata, packages) { id, devId ->
-            !isDevDependencyOf(id, devId, metadata)
+        val projectId = metadata["workspace_members"]
+            .map { it.textValueOrEmpty() }
+            .single { it.contains(projectName) && it.contains(projectVersion) }
+
+        val projectNode = metadata["packages"].single { it["id"].textValueOrEmpty() == projectId }
+
+        val directDependencies = projectNode["dependencies"].filter { node ->
```

Comment:
Nit: See above, maybe invert the logic here.

Reply:
Done.